### PR TITLE
Add table azure_storage_share_file closes #455

### LINKED ISSUE #455: Add table azure_storage_share_file
**References**
https://docs.microsoft.com/de-de/rest/api/storagerp/file-shares/list

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/azure_storage_share_file []

PRETEST: tests/azure_storage_share_file

TEST: tests/azure_storage_share_file
Running terraform
data.azurerm_client_config.current: Refreshing state...
data.null_data_source.resource: Refreshing state...
azurerm_resource_group.named_test_resource: Creating...
azurerm_resource_group.named_test_resource: Creation complete after 4s [id=/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest15836]
azurerm_storage_account.named_test_resource: Creating...
azurerm_storage_account.named_test_resource: Still creating... [10s elapsed]
azurerm_storage_account.named_test_resource: Still creating... [20s elapsed]
azurerm_storage_account.named_test_resource: Still creating... [30s elapsed]
azurerm_storage_account.named_test_resource: Creation complete after 32s [id=/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest15836/providers/Microsoft.Storage/storageAccounts/turbottest15836]
azurerm_storage_share.named_test_resource: Creating...
azurerm_storage_share.named_test_resource: Creation complete after 2s [id=https://turbottest15836.file.core.windows.net/turbottest15836]
azurerm_storage_share_file.named_test_resource: Creating...
azurerm_storage_share_file.named_test_resource: Creation complete after 2s [id=https://turbottest15836.file.core.windows.net/turbottest15836//turbottest15836]

Warning: Deprecated Resource

  on variables.tf line 31, in data "null_data_source" "resource":
  31: data "null_data_source" "resource" {

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 4 added, 0 changed, 0 destroyed.

Outputs:

resource_aka = azure:///subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest15836/providers/Microsoft.Storage/storageAccounts/turbottest15836/fileServices/default/shares/turbottest15836
resource_aka_lower = azure:///subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourcegroups/turbottest15836/providers/microsoft.storage/storageaccounts/turbottest15836/fileservices/default/shares/turbottest15836
resource_id = /subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest15836/providers/Microsoft.Storage/storageAccounts/turbottest15836/fileServices/default/shares/turbottest15836
resource_name = turbottest15836

Running SQL query: test-get-query.sql
[
  {
    "id": "/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest15836/providers/Microsoft.Storage/storageAccounts/turbottest15836/fileServices/default/shares/turbottest15836",
    "name": "turbottest15836",
    "storage_account_name": "turbottest15836"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "id": "/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest15836/providers/Microsoft.Storage/storageAccounts/turbottest15836/fileServices/default/shares/turbottest15836",
    "name": "turbottest15836"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "azure:///subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest15836/providers/Microsoft.Storage/storageAccounts/turbottest15836/fileServices/default/shares/turbottest15836",
      "azure:///subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourcegroups/turbottest15836/providers/microsoft.storage/storageaccounts/turbottest15836/fileservices/default/shares/turbottest15836"
    ],
    "title": "turbottest15836"
  }
]
✔ PASSED

POSTTEST: tests/azure_storage_share_file

TEARDOWN: tests/azure_storage_share_file

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```sql
select
  name,
  storage_account_name,
  type,
  access_tier,
  share_quota,
  enabled_protocols
from
  azure_storage_share_file;
```

```
+-----------------------------------------------------+----------------------+-------------------------------------------------------+----------------------+-------------+-------------------+
| name                                                | storage_account_name | type                                                  | access_tier          | share_quota | enabled_protocols |
+-----------------------------------------------------+----------------------+-------------------------------------------------------+----------------------+-------------+-------------------+
| cs-partha-turbotad-onmicrosoft-com-1003200052b74987 | csg1003200052b74987  | Microsoft.Storage/storageAccounts/fileServices/shares | TransactionOptimized | 6           | SMB               |
| test1                                               | tetsg8900000         | Microsoft.Storage/storageAccounts/fileServices/shares | TransactionOptimized | 512         | SMB               |
+-----------------------------------------------------+----------------------+-------------------------------------------------------+----------------------+-------------+-------------------+

```

```sql
 select
  name,
  storage_account_name,
  type,
  access_tier,
  share_quota,
  enabled_protocols
from
  azure_storage_share_file
where
  type = 'Microsoft.Storage/storageAccounts/fileServices/shares';
```

```
+-----------------------------------------------------+----------------------+-------------------------------------------------------+----------------------+-------------+-------------------+
| name                                                | storage_account_name | type                                                  | access_tier          | share_quota | enabled_protocols |
+-----------------------------------------------------+----------------------+-------------------------------------------------------+----------------------+-------------+-------------------+
| cs-partha-turbotad-onmicrosoft-com-1003200052b74987 | csg1003200052b74987  | Microsoft.Storage/storageAccounts/fileServices/shares | TransactionOptimized | 6           | SMB               |
| test1                                               | tetsg8900000         | Microsoft.Storage/storageAccounts/fileServices/shares | TransactionOptimized | 512         | SMB               |
+-----------------------------------------------------+----------------------+-------------------------------------------------------+----------------------+-------------+-------------------+
```

```sql
select
  name,
  storage_account_name,
  type,
  access_tier,
  access_tier_change_time,
  share_quota,
  enabled_protocols
from
  azure_storage_share_file
where
  access_tier = 'TransactionOptimized';
```

```
+-----------------------------------------------------+----------------------+-------------------------------------------------------+----------------------+-------------------------+-------------+-----
| name                                                | storage_account_name | type                                                  | access_tier          | access_tier_change_time | share_quota | enab
+-----------------------------------------------------+----------------------+-------------------------------------------------------+----------------------+-------------------------+-------------+-----
| cs-partha-turbotad-onmicrosoft-com-1003200052b74987 | csg1003200052b74987  | Microsoft.Storage/storageAccounts/fileServices/shares | TransactionOptimized | <null>                  | 6           | SMB 
| test1                                               | tetsg8900000         | Microsoft.Storage/storageAccounts/fileServices/shares | TransactionOptimized | <null>                  | 512         | SMB 
+-----------------------------------------------------+----------------------+-------------------------------------------------------+----------------------+-------------------------+-------------+-----

```

```sql
select
  name,
  storage_account_name,
  type,
  access_tier,
  access_tier_change_time,
  share_quota,
  enabled_protocols
from
  azure_storage_share_file
order by share_quota desc limit 1;
```

```
+-------+----------------------+-------------------------------------------------------+----------------------+-------------------------+-------------+-------------------+
| name  | storage_account_name | type                                                  | access_tier          | access_tier_change_time | share_quota | enabled_protocols |
+-------+----------------------+-------------------------------------------------------+----------------------+-------------------------+-------------+-------------------+
| test1 | tetsg8900000         | Microsoft.Storage/storageAccounts/fileServices/shares | TransactionOptimized | <null>                  | 512         | SMB               |
+-------+----------------------+-------------------------------------------------------+----------------------+-------------------------+-------------+-------------------+
```

</details>

### CODE REVIEW

File: docs/tables/azure_storage_share_file.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Table: azure_storage_blob
```
Comment (bigdatasourav) [inline docs/tables/azure_storage_share_file.md:1]:
```suggestion
# Table: azure_storage_share_file
```

File: azure/table_azure_storage_share_file.go [hunk 2]
```diff
@@ -0,0 +1,277 @@
+package azure
+
+import (
+	"context"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2019-06-01/storage"
```
Comment (bigdatasourav) [inline azure/table_azure_storage_share_file.go:7]: The current version for storage is 2021-04-01, can we try with it?

Comment (ParthaI) [reply azure/table_azure_storage_share_file.go:7]: With version `github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2021-04-01/storage`, the get call has been removed.

**Review by bigdatasourav — APPROVED** [review]
LGTM

**Pull request merged**